src/main/java/net/glowstone/io/anvil/RegionFile.java

```diff
@@ -200,19 +214,19 @@ public DataInputStream getChunkDataInputStream(int x, int z) throws IOException
         if (version == VERSION_GZIP) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new GZIPInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(data))));
```

Comment:
But GZIPInputStream is the bottleneck in this case. The speed of using the compressed data is what is ultimately limiting the speed, not subsequently using the uncompressed data. You might see a small increase in performance by buffering the uncompressed data, but will it be worth it?

Because keep in mind that having a buffer for the compressed data is buffering more data in less bytes than the uncompressed stream.

I think it's better to tune the included buffers rather than adding a new `BufferedInputStream` on top of it, especially when you have its buffer for the uncompressed data much larger than than the inside buffers for the compressed data.